flink-connectors/flink-connector-aws/src/main/java/org/apache/flink/streaming/connectors/kinesis/unified/util/AWSConfigConstants.java

```diff
@@ -0,0 +1,167 @@
+
+/** Configuration keys for AWS service usage. */
+@PublicEvolving
+public class AWSConfigConstants {
+
+    /**
+     * Possible configuration values for the type of credential provider to use when accessing AWS
+     * Kinesis. Internally, a corresponding implementation of {@link AWSCredentialsProvider} will be
+     * used.
+     */
+    public enum CredentialProvider {
+
+        /**
+         * Look for the environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY to create
+         * AWS credentials.
+         */
+        ENV_VAR,
+
+        /**
+         * Look for Java system properties aws.accessKeyId and aws.secretKey to create AWS
+         * credentials.
+         */
+        SYS_PROP,
+
+        /** Use a AWS credentials profile file to create the AWS credentials. */
+        PROFILE,
+
+        /**
+         * Simply create AWS credentials by supplying the AWS access key ID and AWS secret key in
+         * the configuration properties.
+         */
+        BASIC,
+
+        /**
+         * Create AWS credentials by assuming a role. The credentials for assuming the role must be
+         * supplied. *
+         */
+        ASSUME_ROLE,
+
+        /**
+         * Use AWS WebIdentityToken in order to assume a role. A token file and role details can be
+         * supplied as configuration or environment variables. *
+         */
+        WEB_IDENTITY_TOKEN,
+
+        /**
+         * A credentials provider chain will be used that searches for credentials in this order:
+         * ENV_VARS, SYS_PROPS, WEB_IDENTITY_TOKEN, PROFILE in the AWS instance metadata. *
+         */
+        AUTO,
+    }
+
+    /** The AWS region of the Kinesis streams to be pulled ("us-east-1" is used if not set). */
+    public static final String AWS_REGION = "aws.region";
+
+    /**
+     * The credential provider type to use when AWS credentials are required (BASIC is used if not
+     * set).
+     */
+    public static final String AWS_CREDENTIALS_PROVIDER = "aws.credentials.provider";
+
+    /** The AWS access key ID to use when setting credentials provider type to BASIC. */
+    public static final String AWS_ACCESS_KEY_ID = accessKeyId(AWS_CREDENTIALS_PROVIDER);
+
+    /** The AWS secret key to use when setting credentials provider type to BASIC. */
+    public static final String AWS_SECRET_ACCESS_KEY = secretKey(AWS_CREDENTIALS_PROVIDER);
+
+    /** Optional configuration for profile path if credential provider type is set to be PROFILE. */
+    public static final String AWS_PROFILE_PATH = profilePath(AWS_CREDENTIALS_PROVIDER);
+
+    /** Optional configuration for profile name if credential provider type is set to be PROFILE. */
+    public static final String AWS_PROFILE_NAME = profileName(AWS_CREDENTIALS_PROVIDER);
+
+    /**
+     * The role ARN to use when credential provider type is set to ASSUME_ROLE or
+     * WEB_IDENTITY_TOKEN.
+     */
+    public static final String AWS_ROLE_ARN = roleArn(AWS_CREDENTIALS_PROVIDER);
```

Comment:
Some of the fields seem to be not used. Can you clarify a bit more the purpose of this class and maybe update the doc string?

Reply:
Thank you, I will remove the unused fields. But I wanted to ask about this actually, there is some duplication here with `AWSConfigConstants` and `AwsV2Util` of `flink-connector-kinesis`. I can't use the same trick we used for `KinesaliteContainer` since it's not a test class (for fear of breaking existing customer code).